rfc/rfc-79/rfc-79-2.md

```diff
@@ -0,0 +1,99 @@
+
+![mutable table service lifecycle (3)](https://github.com/user-attachments/assets/7c7d67f5-b6d3-44e6-990b-340920bc16a7)
+
+Once cancellable table service plans are made mutable in this manner, clean can rollback failed cancellable table service plans that have met the cancellation-policy critera, similar to how clean currently rolls back failed ingestion writes. Specifically, clean can check for any failed cancelled table service plans that are already part of a pending rollback plan or meet the cancellation-policy. From there a rollback can be scheduled/executed for each instant.
+With these changes, a failed cancellable table service plan that has met its cancellation policy will be guaranteed to be attempted for rollback by the next clean. If a user wants to immeditaly cleanup a failed cancellable plan, they can bypass the cancellation policy by scheduling and executing a rollback plan, the same way that clean will cleanup these plans.
+
+This meets the critera for goal (2). But comes with the following drawback:
+* The instant metadata file for the cancellable table service plan will be deleted on rollback, analogous to how rollback of a ingestion instant works. This can make it more difficult to debug failed/stuck cancellable table service plans
+  
+#### (B) Adding a cancel operation/state for cancellable plans
+An alternate approach can involve updating the possible tmeline actions / states, by making the following changes:
+* Add an ".aborted" state type for cancellable table service plan. 
+* Add a new action type "cancel" with two states ".cancel.requested" and ".cancel". The  ".cancel.requested" metadata file will be a plan that targets a (cancellable table service) instant. Once said instant is transitioned to aborted state, the action can be completed and transitioned to ".cancel"
+
+A new cancel API will be added that a writer can use to target a cancellable table service plan to be aborted. It will create a cancel.request plan for the target instant, and execute it. If an existing cancel.requested plan for the target already exists, it will try to execute that directly (similar to how the rollback API handles pending rollbacks). Execution of a cancel action involves the followings steps
+1. Rollback the instant without deleting the table service plan.
+2. Transition the table service instant to .aborted, if it hasn't been already
+3. Transition the cancel plan to .cancel
+Once the cancel action has been transitioned to ".cancel", it can be considered complete. The reason this cancel action needs a ".requested" state is in order to allow clean/archival to be able to infer when a cancel action is completed.
```

Comment:
lets sync up f2f. I feel, we can further simplify this.

Reply:
Sure. Another way I had in mind was adjusting this approach to not need an "aborted" state but instant have an "empty" commit. For example, upon being cancelled and rolled back, a clustering instant would still be completed with a ".repalcecommit" but wouldn't actually mark any files as replaced in the metadata. But I felt this might be confusing to users and would cause replacecommit.requested not match the replacecommit file metadata